app/controllers/snippets_controller.rb

```diff
@@ -77,8 +77,9 @@ def set_vault
     end
 
     def set_snippet
-      @vault = Vault.find(params[:vault_id])
-      @snippet = @vault.snippets.find(params[:id])
+      @vault = Vault.find_by_id(params[:vault_id])
```

Comment:
This is what i came up with. no other choice but to query a new @vault  since in snippet controller have `vault_id` while in vault controller have `id`. i ended up removing set_vault in both controller.
```
app/controllers/application_controller.rb

  def set_vault
    @vault = params[:vault_id] ? Vault.find_by_id(params[:vault_id]) : Vault.find_by_id(params[:id])
    redirect_to root_url if @vault.nil? || @vault.user_username != params[:username]
  end
```